FWCore/Utilities/interface/Range.h

```diff
@@ -16,6 +18,12 @@ namespace edm {
     T end() const { return end_; }
 
     bool empty() const { return begin_ == end_; }
+    auto size() const { return end_ - begin_; }
+
+    auto const& operator[](std::size_t idx) const { return *(begin_ + idx); }
+
+    auto const& front() const { return *begin_; }
+    auto const& back() const { return *(end_ - 1); }
```

Comment:
These additions make `edm::Range` to essentially require random-access iterators. They also make `edm::Range` to look more like C++20 [`std::span`](https://en.cppreference.com/w/cpp/container/span) than the C++20 [`std::ranges::range`](https://en.cppreference.com/w/cpp/ranges/range) concept. How about leaving `edm::Range` as it is and create new `edm::Span` with these additions (that would later be replaced with `std::span`)?

Reply:
Absolutely! I also realized that it now requires random-access iterators to work efficiently, but thought that was okay since we only use `edm::Range` for those (so far). Your idea of making it consistent with `std` is really nice, thank you. I'll do that.